R/modify-model-field.R

```diff
@@ -0,0 +1,233 @@
+#####################
+# Modifying models
+#####################
+
+#' Modify field in model object
+#'
+#' Implementation function for updating fields in a `bbi_{.model_type}_model` object
+#' Also checks the object against the corresponding YAML before modifying (and errors if they are out of sync) and then writes the modified object back to the YAML.
+#' @param .mod The `bbi_{.model_type}_model` object to modify
+#' @param .field Character scaler of the name of the component to modify
+#' @param .value Whatever is to be added to `.mod[[.field]]`, typically a character scaler or vector
+#' @param .append Boolean for whether to concatenate new values with currently present values. TRUE by default. If FALSE, new values will overwrite old values.
+#' @param .unique Boolean for whether to de-duplicate `.mod[[.field]]` after adding new values. TRUE by default.
+#' @rdname modify_model_field
+#' @export
+modify_model_field <- function(.mod, .field, .value, .append = TRUE, .unique = TRUE) {
+
+  # update .mod with any changes from yaml on disk
+  check_yaml_in_sync(.mod)
+
+  # Either append new value or overwrite with new value
+  if (isTRUE(.append)) {
+    .mod[[.field]] <- c(.mod[[.field]], .value)
+  } else {
+    .mod[[.field]] <- .value
+  }
+
+  # de-duplicate values
+  if (isTRUE(.unique)) {
+    .mod[[.field]] <- .mod[[.field]] %>% unique()
```

Comment:
I'll go into this more in a tech session, but its best to avoid pipes unless needed in programmatic contexts

```suggestion
    .mod[[.field]] <- unique(.mod[[.field]])
```

that is because the pipe is an extra function call, which means it adds an extra layer to stack traces as well as also makes profiling harder to digest

Reply:
That makes sense. I have this pattern in... a lot of places.
```
$ grep -r \%\>\% rbabylon/R | wc -l
      64
```
We can talk at tech session and I'll make a cleanup PR if you think it's worthwhile. If not, I'll just keep this in mind moving forward.